src/System.IO.Ports/src/System/IO/Ports/SerialStream.Win32.cs

```diff
@@ -18,5 +18,15 @@ public SafeFileHandle OpenPort(uint portNumber)
                 FileMode.Open,  // comm devices must use OPEN_EXISTING
                 Interop.Kernel32.FileOperations.FILE_FLAG_OVERLAPPED);
         }
+
+        public SafeFileHandle OpenPort(string portName)
+        {
+            // portName: eg "\\?\USB#VID_1B4F&PID_214F&MI_00#6&20b7e7e3&0&0000#{86e0d1e0-8089-11d0-9ce4-08003e301f73}"
```

Comment:
If port name starts with `\\?` won't you end up with double `\\?` in the native call?

cc: @JeremyKuhne

Reply:
We used to explicitly [disallow](https://referencesource.microsoft.com/#System/sys/system/io/ports/SerialPort.cs,417) `\\` but that wasn't necessary on any Windows that is remotely modern. We sort of still "block" it by putting the prefix on.

Aren't we no longer building UAP? If so, I'd fold this back in to the .Windows and get rid of the UAP'isms. If not, I'd change both to take "portName" and let the UAP code do the parsing for a number. I don't think we should be validating the names at all.

The right way to do this is call `Path.GetFullPath(portName)`, which will put the `\\.\` on `COM1`. Users should be specifying full device paths if they want direct access to other device names. Don't add the `\\?\` or `\\.\` here. We should give the full device name in the list of ports if they're not "legacy" `COMx` names.

Reply:
Sorry, double checked this, and my comment was wrong, included the preamble '\\?\'
But sound not have, Portnames retrived are just

> USB#VID_1B4F&PID_214F&MI_00#6&381731fd&0&0000#{86e0d1e0-8089-11d0-9ce4-08003e301f73}